specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/batchInference.json

```diff
@@ -0,0 +1,1004 @@
+        "AppendRow"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "BatchOutputAction",
+        "modelAsString": false
+      }
+    },
+    "BatchOutputConfiguration": {
+      "description": "Batch inference output configuration.",
+      "type": "object",
+      "properties": {
+        "outputAction": {
+          "$ref": "#/definitions/BatchOutputAction"
+        },
+        "appendRowFileName": {
+          "description": "Customized output file name for append_row output action.",
+          "type": "string"
+        }
+      }
+    },
+    "InferenceEndpointKeysResource": {
+      "description": "Inference endpoint keys.",
+      "type": "object",
+      "properties": {
+        "primaryKey": {
+          "description": "Primary key.",
+          "type": "string"
+        },
+        "secondaryKey": {
+          "description": "Secondary key.",
+          "type": "string"
+        }
+      }
+    },
+    "DeploymentScaleSettings": {
+      "type": "object",
+      "properties": {
+        "nodeCount": {
+          "format": "int32",
+          "description": "Number of Compute nodes to be used for running the deployment",
+          "type": "integer"
+        }
+      }
+    },
+    "DeploymentProvisioningState": {
+      "enum": [
+        "Creating",
+        "Deleting",
+        "Scaling",
+        "Updating",
+        "Succeeded",
+        "Failed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "DeploymentProvisioningState",
+        "modelAsString": false
+      }
+    },
+    "EndpointProvisioningState": {
+      "description": "State of endpoint provisioning.",
+      "enum": [
+        "Creating",
+        "Deleting",
+        "Succeeded",
+        "Failed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "EndpointProvisioningState",
+        "modelAsString": false
+      }
+    },
+    "EndpointAuthModeType": {
+      "enum": [
+        "AMLTokenAuth",
+        "KeyAuth"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "EndpointAuthModeType",
+        "modelAsString": false
+      }
+    },
+    "BatchDeployment": {
+      "description": "Batch inference settings per deployment.",
+      "type": "object",
+      "properties": {
+        "scaleSettings": {
+          "$ref": "#/definitions/DeploymentScaleSettings"
+        },
+        "batchSettings": {
+          "$ref": "#/definitions/BatchDeploymentSettings"
+        },
+        "tags": {
+          "description": "Tag dictionary. Tags can be added, removed, and updated.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "create",
+            "read",
+            "update"
+          ]
+        },
+        "description": {
+          "description": "Description of the endpoint deployment.",
+          "type": "string"
+        },
+        "properties": {
```

Comment:
>properties [](start = 9, length = 10)

Same here there are multi level of properties. What is the purpose?

Reply:
There is only 1 level of properties here, the external 'properties' is the resource envelope wrapper?